WIP: Suggestion on how to run new variant CI

DunderRoffe opened the pull request:
This PR is to discuss how to run the CI for the suggested new variant system.
Uses https://github.com/Pelagicore/meta-pelux/pull/427 version of meta-pelux

Commit:
WIP: Draft for use of new variant system

This is a work in progress and relies on a not finialized version of
meta-pelux.

The point of this patch is to showcase one example on how to do the
uncommenting of local.conf and bblayers.conf which is required by the
suggested change to handle variants.

It requires new types of meta data to be sent to the CI, namely machine
and a list of features to enable. However, it does not propose a means
or format of doing so.

Commit:
buildImage: remove unused parameter variantName

Commit:
yocot2: Remove unused function buildWithLayer

Commit:
Select Machine and features in bash

Use a bash script to select machine and features instead of doing it in
groovy. This makes it possible for developers to use this script locally
as well. Meaning the CI is using the same things as is available to
developers.

pookimon commented on ci-scripts/yocto2.groovy:
```diff
@@ -180,7 +180,12 @@ String getStringEnvVar(String envVarName, String defaultValue) {
+//
+// "features" is a list of strings representing additional features in the build.
+// Example: features = ["qtauto"]
+void buildManifest(String machine, List features, String imageName, String layerToReplace="", String newLayerPath="") {
```
Main file for Jenkins is [https://github.com/Pelagicore/pelux-manifests/blob/new_variant_ci/Jenkinsfile](Jenkinsfile) buildManifest call should be changed there as well

DunderRoffe replied:
good catch

DunderRoffe replied:
should be fixed now

Commit:
WIP: Copy multiconfig from templateconf

This is a required step when using meta-acrn which require additional
configuration files which the oe-init-build-env script does not copy.

DunderRoffe commented:
@pookimon Jenkins wont build for some reason... Is this because of naming it WIP?

pookimon commented:
buildManifest now need 3 parameters machine,  features,  imageName,

DunderRoffe commented:
@JEderonn @martin-ejdestig suggestion is to format variant name using

```python
String variantName = machine_name + " " + features.join(" ")
```
Example:
Using machine = "intel-corei7-64", features = [qtauto, acrn]
-> variant = "intel-corei7-64 qtauto acrn"

DunderRoffe commented:
> buildManifest now needs 3 parameters machine, features, imageName,

Well it does not even attempt to build. Says "disabled" in Jenkins :(

Commit:
fixup! WIP: Draft for use of new variant system

Commit:
fixup! WIP: Draft for use of new variant system

Commit:
fixup! Select Machine and features in bash

Commit:
fixup! WIP: Draft for use of new variant system

JEderonn commented on Jenkinsfile:
```diff
@@ -61,7 +61,14 @@ try {
     variantMap["${list[0]}"] = {
-        buildOnYoctoNode(list[0], list[1])
+
+        // TODO get these values from variantMap (or similar).
```
I assume this also needs to be fixed before merging.

Today `VARIANT_LIST` is used. Syntax is a list of items of format `<variant>:<image>`

Any ideas of the new format?
`<machine>:[<feature>:][<feature>:][<feature>:]<image>`

DunderRoffe replied:
I like the format. however : is not a good separator since it is used by bitbake multiconfig and will therefore cause issues when building using meta-acrn.

So yes to the format with a bitbake safe separator such as "_" or "|"

JEderonn replied:
I prefer `|` or `#`, e.g.
`rcarh3#qtas#acrn`
`rcarh3|qtas|acrn`

The `#` might be more readable than `|`

pookimon replied:
I vote for "|"

DunderRoffe commented:
Closing due to inactivity
